### PR TITLE
Separately store acceptable RPKI data.

### PR DESCRIPTION
**Pull request by partim**
This PR introduces the concept of a _store_ that maintains a set of RPKI data from accepted publication points, ie., those with a valid manifest and all objects on the manifest present and correctly hashed. The store is updated from a _cache_, which synchronizes data with upstream repositories if they have new data this is acceptable.

### CODE REVIEW

**Comment by partim** [comment]
Things still left to do:

- [ ] Let `Validation` keep its cache and store.
- [ ] If update is disabled, let `Validation` not have a cache and remove the no-update capability from the cache.
- [ ] Add cleanup.

**Comment by partim** [comment]
Note: The RRDP cache has only been minimally adjusted to work with the new APIs where necessary. It will receive either a rewrite or cleanup as a separate PR.

**Review by density215 — APPROVED** [review]
I pushed a commit with a few typos. That's all.

File: src/store.rs [hunk 1]
```diff
@@ -0,0 +1,1069 @@
...
+use crate::metrics::Metrics;
+
+
+//------------ Store ---------------------------------------------------------
+
+/// A store for correctly published RPKI objects.
+///
+/// The store retains a copy of curated, published RPKI data. Its intended use
+/// is for keeping most recent data of a given RPKI publication point that was
+/// found to be correctly published. However, the store doesn’t enforce this,
```
[the remark this answers is not in the document]
Comment (partim) [reply src/store.rs]: ‘the most recent’.

**Pull request merged**